src/runtime_strings.js

```diff
@@ -188,15 +188,19 @@ function stringToUTF8(str, outPtr, maxBytesToWrite) {
 // Returns the number of bytes the given Javascript string takes if encoded as a UTF8 byte array, EXCLUDING the null terminator byte.
 function lengthBytesUTF8(str) {
   var len = 0;
-  for (var i = 0; i < str.length; ++i) {
+  for (var i = 0, l = str.length; i < l; ++i) {
```

Comment:
Does this really help in JS to play these kinds of tricks?

It looks like it costs a few bytes in code size, and an extra JS variable, in exchange for some performance?   If we are going to do this we should probably be consistent throughout the codebase.

Reply:
It usually increases speed of cold start, but I can revert this line if you prefer lower code size